SWIG/integrals.i

```diff
@@ -133,41 +152,57 @@ class GaussHermiteIntegration {
     GAUSSIAN_QUADRATURE_METHODS;
 };
 
-class GaussJacobiIntegration {
+class GaussJacobiIntegration: public GaussianQuadrature {
   public:
     GaussJacobiIntegration(Size n, Real alpha, Real beta);
     GAUSSIAN_QUADRATURE_METHODS;
 };
 
-class GaussHyperbolicIntegration {
+class GaussHyperbolicIntegration: public GaussianQuadrature {
   public:
     GaussHyperbolicIntegration(Size n);
     GAUSSIAN_QUADRATURE_METHODS;
 };
 
-class GaussLegendreIntegration {
+class GaussLegendreIntegration: public GaussianQuadrature {
   public:
     GaussLegendreIntegration(Size n);
     GAUSSIAN_QUADRATURE_METHODS;
 };
 
-class GaussChebyshevIntegration {
+class GaussChebyshevIntegration: public GaussianQuadrature {
   public:
     GaussChebyshevIntegration(Size n);
     GAUSSIAN_QUADRATURE_METHODS;
 };
 
-class GaussChebyshev2ndIntegration {
+class GaussChebyshev2ndIntegration: public GaussianQuadrature {
   public:
     GaussChebyshev2ndIntegration(Size n);
     GAUSSIAN_QUADRATURE_METHODS;
 };
 
-class GaussGegenbauerIntegration {
+class GaussGegenbauerIntegration: public GaussianQuadrature {
   public:
     GaussGegenbauerIntegration(Size n, Real lambda);
     GAUSSIAN_QUADRATURE_METHODS;
 };
 
 
+#ifdef QL_BOOST_HAS_TANH_SINH
```

Comment:
This is defined in the C++ headers but we don't have this available in the `.i` files.  I would export it unconditionally and rely on the runtime error in the fallback implementation.